src/ansi-converter.ts

```diff
@@ -0,0 +1,100 @@
+import * as vscode from "vscode";
+import AnsiColorsParser, { ParsedColor, ParsedSpan, parse } from "ansicolor";
+
+// taken from the ansicolor source code
+const colorNameAsBright = {
+  'red': 'lightRed',
+  'green': 'lightGreen',
+  'yellow': 'lightYellow',
+  'blue': 'lightBlue',
+  'magenta': 'lightMagenta',
+  'cyan': 'lightCyan',
+  'black': 'darkGray',
+  'lightGray': 'white'
+}
+
+export function doesTextContainAnsiCodes(text: string): boolean {
+  return AnsiColorsParser.isEscaped(text);
+}
+
+export function convertAnsiTextToHtml(text: string): string {
+  const parsed = parse(text).spans;
+
+  // According to VS Code source code, only span supports style
+  // https://github.com/microsoft/vscode/blob/6d2920473c6f13759c978dd89104c4270a83422d/src/vs/base/browser/markdownRenderer.ts#L301
+  const html = parsed.map((span) => `<span style="${getSupportedCSSFromAnsiSpan(span)}">${span.text.replaceAll('\n', '<br/>')}</span>`).join("");
```

Comment:
We should make sure to escape HTML found in the text if this isn't done by the ansicolor library

Reply:
The ansicolor does not escape html in the text, but html is sanitized anyway on vscode side